tiledb/sm/filter/bitshuffle_filter.cc

```diff
@@ -122,44 +126,47 @@ Status BitshuffleFilter::compute_parts(
 }
 
 Status BitshuffleFilter::shuffle_part(
-    const WriterTile&, const ConstBuffer* part, Buffer* output) const {
-  auto tile_type_size = static_cast<uint8_t>(datatype_size(filter_data_type_));
-  auto part_nelts = part->size() / tile_type_size;
-  auto bytes_processed = bshuf_bitshuffle(
-      part->data(), output->cur_data(), part_nelts, tile_type_size, 0);
-
-  switch (bytes_processed) {
-    case -1:
-      return LOG_STATUS(
-          Status_FilterError("Bitshuffle error; Failed to allocate memory."));
-    case -11:
-      return LOG_STATUS(Status_FilterError("Bitshuffle error; Missing SSE."));
-    case -12:
-      return LOG_STATUS(Status_FilterError("Bitshuffle error; Missing AVX."));
-    case -80:
+    Datatype filter_data_type,
```

Comment:
why pass this as an argument? Isn't it the private member variable `filter_data_type_`?